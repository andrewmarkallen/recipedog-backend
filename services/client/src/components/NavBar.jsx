import React, { useState } from 'react'
import { Redirect } from 'react-router-dom'
import { Button, FormControl, Glyphicon, FormGroup, InputGroup, Modal } from 'react-bootstrap'

const SearchModal = (props)  => {

  return(
      <Modal id="search-modal" show={props.show} onHide={props.handleCloseSearch}>
      <Modal.Header closeButton id="search-header">
        <h1>Search by title and tags</h1>
      </Modal.Header>
      <Modal.Body>
        <FormGroup>
          <InputGroup classname="search-group">
            <FormControl bsSize="large" type="text" placeholder="enter search terms..."/>
            <InputGroup.Button>
              <Button bsSize="large"><Glyphicon glyph="search"/></Button>
            </InputGroup.Button>
          </InputGroup>
        </FormGroup>
      </Modal.Body>
    </Modal>
  )
}

const NavBar = (props)  => {

  const[redirect, setRedirect] = useState(false)
  const[showSearch, setShowSearch] = useState(false)

  const handleClick = ()  => {
    console.log("clickety")
    var x = document.getElementById("myTopnav");
      if (x.className === "fixed-nav-bar") {
        x.className += " responsive";
        console.log("adding")
      } else {
        x.className = "fixed-nav-bar";
      }
  }

  const handleCloseSearch = () => { setShowSearch(false)}


  return(
    <div>
      { redirect && <Redirect to="/"/> }
      <SearchModal show={showSearch} handleCloseSearch={handleCloseSearch}/>
      <nav className="fixed-nav-bar" id="myTopnav">
        <div>
        <button className="icon" id="hamburger" onClick={handleClick}>
          <i className="glyphicon glyphicon-menu-hamburger"></i></button>
        <button id="hamburger" onClick={handleClick}>
          <i className="glyphicon glyphicon-menu-hamburger"></i></button>
        <a href="/" className="active">Home</a>
        {props.isAuthenticated && <a href="/myrecipes" >Recipes</a> }
        <a href="/about">About</a>
        {props.isAuthenticated && <a href="/status">User Status</a>}
        {!props.isAuthenticated && <a href="/register">Register</a>}
        {!props.isAuthenticated && <a href="/login">Log In</a>}
        {props.isAuthenticated && <a href="/logout">Log Out</a>}

      </div>
        <div id="navbar-brand" onClick={()=>setRedirect(true)}> RecipeDog</div>
        <button className="icon" onClick={()=>setShowSearch(true)} id="navbar-search">
          <i className="glyphicon glyphicon-search"></i>
        </button>
      </nav>
    </div>

  )
}

export default NavBar
