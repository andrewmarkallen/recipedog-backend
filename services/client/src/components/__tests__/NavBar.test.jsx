import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'

import NavBar from '../NavBar'

const title = 'Hello, World!'

test('NavBar renders properly', ()  => {

  //todo: rewrite test

})

test('NavBar renders a snapshot properly', ()  =>  {
  const tree = renderer.create(
    <Router location="/"><NavBar title={title}/></Router>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
