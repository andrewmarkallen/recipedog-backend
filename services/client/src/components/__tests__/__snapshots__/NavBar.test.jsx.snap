// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders a snapshot properly 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <span
        className="navbar-brand"
      >
        Hello, World!
      </span>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav"
        role={null}
      >
        <li
          className="active"
          role="presentation"
          style={Object {}}
        >
          <a
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="/about"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            About
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="/status"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            User Status
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="/register"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Register
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="/login"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Log In
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="/logout"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Log Out
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
